jdaviz/components/viewer_data_select.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <j-tooltip v-if="menuButtonAvailable()" tipid="viewer-toolbar-data">
+    <v-menu offset-y :close-on-content-click="false" v-model="viewer.data_open">
+      <template v-slot:activator="{ on, attrs }">
+        <v-btn 
+          text 
+          elevation="3" 
+          v-bind="attrs" 
+          v-on="on" 
+          color="white"
+          tile
+          icon
+          outlined
+          :class="{active: viewer.data_open}"
+          style="height: 42px; width: 42px">
+          <v-icon>mdi-format-list-bulleted-square</v-icon>
+        </v-btn>
+      </template>
+  
+      <v-list style="max-height: 500px; width: 350px;" class="overflow-y-auto">
+        <v-row v-for="item in filteredDataItems" :key="item.id" style="padding-left: 25px">
+          <j-tooltip tipid='viewer-data-select-toggle'>
+            <v-btn 
+              icon
+              :color="viewer.selected_data_items.includes(item.id) ? 'accent' : 'default'"
+              @click="$emit('data-item-selected', {
+                id: viewer.id,
+                item_id: item.id,
+                checked: !viewer.selected_data_items.includes(item.id)
+              })">
+                <v-icon>{{viewer.selected_data_items.includes(item.id) ? "mdi-eye" : "mdi-eye-off"}}</v-icon>
+            </v-btn>
+          </j-tooltip>
+
+          <span class='text--primary' style="overflow-wrap: anywhere; font-size: 12pt; padding-top: 6px; padding-left: 6px">
+            {{item.name}}
+          </span>
+
+          <div v-if="isDeletable(item)" style="position: absolute; right: 10px">
+            <j-tooltip tipid='viewer-data-select-delete'>
+              <v-btn
+                icon
+                @click="$emit('data-item-remove', {item_name: item.name})"
+              ><v-icon>mdi-delete</v-icon></v-btn>
+            </j-tooltip>
+          </div>
+        </v-row>
+      </v-list>
+
+    </v-menu>
+  </j-tooltip>
+</template>
+<script>
+
+module.exports = {
+  props: ['data_items', 'viewer', 'app_settings'],
+  methods: {
+    menuButtonAvailable() {
+      if (this.$props.viewer.config === 'mosviz') {
+        if (this.$props.viewer.reference !== 'spectrum-viewer') {
+          // if making available for other viewers, those cases will also
+          // need to be handled in filteredDataItems below.
+          return false
+        }
+      }
+      return true
+    },
+    isDeletable(item) {
+      if (this.$props.viewer.config === 'mosviz' && item.meta.Plugin === undefined) {
+        return false
+      }
```

Comment:
Maybe we have to ask POs... 😬

Reply:
This has since been changed to only show the delete button for plugin outputs - for now.  If we want to enable deleting of other data entries, we'll need to think about how to handle linking, etc, and how to prevent the app from getting in an unusable state (perhaps by forbidding deleting the reference data, or the last entry, etc).